addons/hr_holidays/static/src/dashboard/time_off_card.js

```diff
@@ -1,17 +1,25 @@
 /* @odoo-module */
 
 import { usePopover } from "@web/core/popover/popover_hook";
+import { useNewAllocationRequest } from '../views/hooks';
+
 
 const { Component } = owl;
 
-export class TimeOffCardPopover extends Component {}
+export class TimeOffCardPopover extends Component {
+    newAllocationRequest() {
+        this.props.onClickNewAllocationRequest()
+    }
```

Comment:
you can do `t-on-click="props.onClickNewAllocationRequest"` directly in the template and you won't need to register a new method here